scripts/clang-tidy-git.sh

```diff
@@ -0,0 +1 @@
+run-clang-tidy-14 -quiet -extra-arg=-std=c++11 -extra-arg=-Wno-unknown-warning-option `git diff --name-only origin/HEAD`
```

Comment:
I don't have `run-clang-tidy-14` just `clang-tidy`.  Can we just default to the later, which may point to a different versions of clang-tidy on different machines?